book/src/development/lint_passes.md

```diff
@@ -0,0 +1,136 @@
+# Lint passes
+
+Before working on the logic of a new lint, there is an important decision
+that every Clippy developers must make: to use
+[`EarlyLintPass`][early_lint_pass] or [`LateLintPass`][late_lint_pass].
+
+In short, the `LateLintPass` has access to type information while the
```

Comment:
Not only type information but also `resolve`: In an early pass, you usually have no information what a symbol means, so most early lints are plainly syntactic.

Reply:
Fixed. (Talks about that in the introduction and in the "Early Lint Pass" section)